src/main/java/org/springframework/integration/kafka/dsl/KafkaProducerMessageHandlerSpec.java

```diff
@@ -195,6 +195,29 @@
 	}
 
 	/**
+	 * Configure a SpEL expression to determine the timestamp at runtime against a
+	 * request Message as a root object of evaluation context.
+	 *
+	 * @param timestampExpression the timestamp expression to use.
+	 * @return the spec.
+	 */
+	public KafkaProducerMessageHandlerSpec<K, V> timestampExpression(String timestampExpression) {
+		return this.timestampExpression(PARSER.parseExpression(timestampExpression));
+	}
+
+	/**
+	 * Configure an {@link Expression} to determine the timestamp at runtime against a
+	 * request Message as a root object of evaluation context.
+	 * @param timestampExpression the timestamp expression to use.
+	 * @return the spec.
+	 */
+	public KafkaProducerMessageHandlerSpec<K, V> timestampExpression(Expression timestampExpression) {
```

Comment:
Please, add also a `Function` variant.
See `partitionId(Function<Message<P>, Integer> partitionIdFunction)` for sample.

Also new API must be marked with `@since`: https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.adoc#follow-the-code-style
Just in case you would like to to avoid similar comments in the future contributions and let us save some time during merge 😉